lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -45,10 +51,33 @@ def on_block(node)
         private
 
         def unused_let_bang(node)
-          let_bang(node) do |method_send, method_name|
+          child_let_bang(node) do |method_send, method_name|
             yield(method_send) unless method_called?(node, method_name)
           end
         end
+
+        def child_let_bang(node)
+          child_block_until_group(node) do |child|
+            node_with_name = let_bang(child)
+
+            yield node_with_name if node_with_name
+          end
+        end
+
+        def child_block_until_group(node, &blk)
+          return unless node.body
+
+          unwrap_nested_begin(node).each_child_node(:block) do |child|
+            next if example_or_shared_group_or_including?(child)
+
+            yield child
+            child_block_until_group(child, &blk)
+          end
+        end
+
+        def unwrap_nested_begin(node)
+          node.body.begin_type? ? node.body : node
+        end
```

Comment:
Just realized the whole implementation of `child_let_bang` could be replaced with `RuboCop::RSpec::ExampleGroup`:

```ruby
RuboCop::RSpec::ExampleGroup.new(node).lets.map { |let|  let_bang(let) }.compact
```

https://github.com/rubocop-hq/rubocop-rspec/blob/master/lib/rubocop/rspec/example_group.rb

I am wondering whether it makes sense to check this approach.

Reply:
Absolutely. Those helpers keep being under the radar.

Reply:
Done. Looks pretty good.